crypto/store/loader_file.c

```diff
@@ -1564,7 +1564,8 @@ static OSSL_STORE_INFO *file_load(OSSL_STORE_LOADER_CTX *ctx,
             pem_free_flag(pem_name, (ctx->flags & FILE_FLAG_SECMEM) != 0, 0);
             pem_free_flag(pem_header, (ctx->flags & FILE_FLAG_SECMEM) != 0, 0);
             pem_free_flag(data, (ctx->flags & FILE_FLAG_SECMEM) != 0, len);
-        } while (matchcount == 0 && !file_eof(ctx) && !file_error(ctx));
+        } while (result == NULL && matchcount == 0
```

Comment:
Why this change?  Has a leak actually been found?  In that case, `file_load_try_decode` should be fixed, as it should *only* return a non-NULL result if `matchcount == 1`

Reply:
As mentioned above, I do not have a witness for an actual mem leak at this point.
All right, since this was just a precaution fix for a potential problem and the change is controversal, I've removed it.